src/js/components/Diagram/Diagram.js

```diff
@@ -222,20 +273,49 @@ const Diagram = forwardRef(({ connections, ...rest }, ref) => {
             colorName = colors[index % colors.length];
           }
 
+          let arrowMarker = null;
+
+          if (typeof endpoint === 'object' && endpoint?.from === 'arrow') {
+            arrowMarker = openArrow(
+              normalizeColor(colorName, theme),
+              index,
+              'openArrowStart',
+              'auto-start-reverse',
+            );
+          } else if (typeof endpoint === 'object' && endpoint?.to === 'arrow') {
+            arrowMarker = openArrow(normalizeColor(colorName, theme), index);
+          } else if (typeof endpoint === 'string' && endpoint === 'arrow') {
+            arrowMarker = (
+              <>
+                {openArrow(
+                  normalizeColor(colorName, theme),
+                  index,
+                  'openArrowStart',
+                  'auto-start-reverse',
+                )}
+                {openArrow(normalizeColor(colorName, theme), index)}
+              </>
+            );
+          }
+
           path = (
-            <path
-              // eslint-disable-next-line react/no-array-index-key
-              key={index}
-              // eslint-disable-next-line react/no-unknown-property
-              animation={animation}
-              {...cleanedRest}
-              stroke={normalizeColor(colorName, theme)}
-              strokeWidth={strokeWidth}
-              strokeLinecap={round ? 'round' : 'butt'}
-              strokeLinejoin={round ? 'round' : 'miter'}
-              fill="none"
-              d={d}
-            />
+            // eslint-disable-next-line react/no-array-index-key
+            <Fragment key={index}>
+              <path
+                // eslint-disable-next-line react/no-unknown-property
+                animation={animation}
+                {...cleanedRest}
+                stroke={normalizeColor(colorName, theme)}
+                strokeWidth={strokeWidth}
+                strokeLinecap={round ? 'round' : 'butt'}
+                strokeLinejoin={round ? 'round' : 'miter'}
+                fill="none"
+                d={d}
+                markerStart={`url("#openArrowStart-${index}")`}
```

Comment:
Is there any downside to having the `markerStart`/`markerEnd` here if the path doesn't have an arrow?

Reply:
Also, if multiple diagrams are rendered on a page, will this index id approach work still?

Reply:
No, there is no downside of having `markerStart` and `markerEnd` over here if the path doesn't have an arrow.
Yes, currently, if multiple diagrams are rendered on page, the index approach will not work. I am working on new approach.